### PR TITLE
make label affinity rules case insensitive

### PR DESCRIPTION
**Pull request by sonchang**

### CODE REVIEW

**Comment by sonchang** [comment]
@alena1108 @cjellick @vincent99 can you guys take a look?

**Comment by sonchang** [comment]
To address: https://github.com/rancherio/rancher/issues/1381

File: code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java [hunk 1]
```diff
@@ -363,7 +363,7 @@ public void setMapDao(GenericMapDao mapDao) {
             .fetchInto(new RecordHandler<Record3<String, String, String>>() {
                 @Override
                 public void next(Record3<String, String, String> record) {
-                    labelKeyValueStatusMap.put(record.value1(), new String[] { record.value2(), record.value3() });
+                    labelKeyValueStatusMap.put(record.value1(), new String[] { record.value2().toLowerCase(), record.value3() });
```
Comment (cjellick) [inline code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java]: no chance that record.value2 can be null?

Comment (sonchang) [inline code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java]: Actually, that could be a possibility.  I'll update

File: code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java [hunk 2]
```diff
@@ -363,7 +363,11 @@ public void setMapDao(GenericMapDao mapDao) {
             .fetchInto(new RecordHandler<Record3<String, String, String>>() {
                 @Override
                 public void next(Record3<String, String, String> record) {
-                    labelKeyValueStatusMap.put(record.value1(), new String[] { record.value2(), record.value3() });
+                    labelKeyValueStatusMap.put(record.value1(),
+                            new String[] {
+                                record.value2() == null ? null : record.value2().toLowerCase(),
```
Comment (alena1108) [inline code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java]: Nit: can use  StringUtils.lowerCase(str) here

**Comment by sonchang** [comment]
Fix nit regarding StringUtils.lowerCase()

**Comment by alena1108** [comment]
LGTM

**Pull request merged**